deployment/terraform/create.go

```diff
@@ -426,16 +427,41 @@ func (t *Terraform) setupProxyServer(extAgent *ssh.ExtAgent) {
 			backends += "server " + addr.PrivateIP + ":8065 max_fails=0;\n"
 		}
 
+		cacheObjects := "10m"
+		cacheSize := "3g"
+		// Extracting the instance class from the type.
+		// Usually they are of the form (m7/c7).(large/2xlarge/4xlarge/..)
+		parts := strings.Split(t.config.ProxyInstanceType, ".")
+		if len(parts) > 1 {
+			// Hack alert. There can be other higher variants like 12xlarge or even metal. But we are keeping it simple for now.
+			switch parts[1] {
+			case "4", "8":
+				cacheObjects = "50m"
+				cacheSize = "16g" // Ideally we'd like half of the total server mem. But the mem consumption rarely exceeds 10G
+				// from my tests. So there's no point stretching it further.
+			}
+		}
```

Comment:
Would it make sense to make this a util? Feels a bit out of place in the whole terraform logic.

Reply:
Hmm .. maybe?